tools/memory_estimator/memory_estimator.py

```diff
@@ -175,25 +251,54 @@ def parse_arguments():
 def main():
     args = parse_arguments()
 
-    # Generate Makefile.
-    generate_makefile(args['lts_path'], args['optimization'], args['lib'])
-
-    # Run make build.
-    warnings = make(args['compiler'])
-
-    # Run make size.
-    calculated_sizes = calculate_size(args['sizetool'])
-
-    # Remove the generated artifacts.
-    if not args['dontclean']:
-        clean()
-        remove_generated_makefile()
-
-    # Print the generated warnings.
-    pretty_print_warnings(warnings)
-
-    # Print the calculated sizes.
-    pretty_print_sizes(calculated_sizes)
+    if args['generate_report']:
+        # Start by creating an empty JSON file which is populated as sizes are
+        # calculated.
+        with open(__GENERATED_JSON_REPORT__, 'w') as json_report_file:
+            json_report_file.write(json.dumps({}))
```

Comment:
Nit: I think we can do away with `json.dumps`:
```python
json_report_file.write('{}')
```

Reply:
This code is no longer needed.